server/api/crud/model_monitoring/deployment.py

```diff
@@ -891,10 +907,19 @@ async def delete_monitoring_function(
                     logger.debug(f"{function_name} pod found, retrying")
                     time.sleep(5)
 
-            v3io_client = v3io.dataplane.Client(endpoint=mlrun.mlconf.v3io_api)
-            stream_paths = server.api.crud.model_monitoring.get_stream_path(
-                project=project, function_name=function_name
+            stream_paths.extend(
+                server.api.crud.model_monitoring.get_stream_path(
+                    project=project, function_name=function_name
+                )
             )
+
+        if stream_paths[0].startswith("v3io"):
+            # Delete V3IO stream
+            import v3io.dataplane
+            import v3io.dataplane.response
```

Comment:
IMO top-level imports are better here, same for `import kafka`.

Reply:
but the whole idea is to avoid importing unnecessary dependencies. There is no reason to import the kafka library if you are using v3io.